server/src/test/java/keywhiz/service/permission/AnyPermissionCheckTest.java

```diff
@@ -0,0 +1,72 @@
+package keywhiz.service.permission;
+
+import java.util.Collections;
+import java.util.List;
+import keywhiz.service.permissions.Action;
+import keywhiz.service.permissions.AnyPermissionCheck;
+import keywhiz.service.permissions.PermissionCheck;
+import org.junit.Rule;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnit;
+import org.mockito.junit.MockitoRule;
+
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+
+public class AnyPermissionCheckTest {
+
+  @Rule public MockitoRule mockito = MockitoJUnit.rule();
+  @Mock private PermissionCheck delegate1;
+  @Mock private PermissionCheck delegate2;
+
+  private static Object source;
+  private static Object target;
+
+  @Test public void isAllowedReturnsFalseWithEmptyCheckList() {
+    PermissionCheck anyPermissionCheck = new AnyPermissionCheck(Collections.emptyList());
+
+    assertFalse(anyPermissionCheck.isAllowed(source, Action.ADD, target));
+  }
+
+  @Test public void isAllowedReturnsFalseWhenAllDelegatesReturnFalse() {
+    PermissionCheck anyPermissionCheck = new AnyPermissionCheck(createDelegatesList(false, false));
+
+    assertFalse(anyPermissionCheck.isAllowed(source, Action.ADD, target));
+  }
+
+  @Test public void isAllowedReturnsTrueWhenOneDelegateReturnsTrue() {
+    PermissionCheck anyPermissionCheck = new AnyPermissionCheck(createDelegatesList(true, false));
+
+    assertTrue(anyPermissionCheck.isAllowed(source, Action.ADD, target));
+  }
+
+  @Test public void checkAllowedOrThrowThrowsExceptionWithEmptyCheckList() {
+    PermissionCheck anyPermissionCheck = new AnyPermissionCheck(Collections.emptyList());
+
+    assertThatThrownBy(() -> anyPermissionCheck.checkAllowedOrThrow(source, Action.ADD, target))
```

Comment:
nit: if you want, you can also check for the exception message (I believe you can call `.hasMessage(...)` after `.isInstanceOf`